feature-libs/checkout/components/components/payment-method/payment-method.component.ts

```diff
@@ -228,8 +228,8 @@ export class PaymentMethodComponent implements OnInit, OnDestroy {
         ? cardLabels.textDefaultPaymentMethod
         : '',
       textBold: paymentDetails.accountHolderName,
-      text: [paymentDetails.cardNumber, cardLabels.textExpires],
-      img: this.getCardIcon(paymentDetails.cardType.code),
+      text: [paymentDetails.cardNumber ?? '', cardLabels.textExpires],
```

Comment:
Could we enforce strict typing without changing the var value with `?? ''` ?

Perhaps use a cast `as string` if nothing else works.

Reply:
This function is for UI display. So, if there is no `cartNumber`, it's better to display an empty string, right?

Reply:
This function is for UI display. So, if there is no `cartNumber`, it's better to display an empty string, right?

Reply:
Indeed, if it's for UI I agree with you.